src/utils/fiatToCrypto.ts

```diff
@@ -0,0 +1,187 @@
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import { Account, AccountStates, GatewayTransactionStates, NotificationTypes, Sdk } from 'etherspot';
+import WertWidget from '@wert-io/widget-initializer';
+import { map } from 'rxjs/operators';
+
+// Services
+import { firebaseRemoteConfig } from 'services/firebase';
+
+// Constants
+import { REMOTE_CONFIG } from 'constants/remoteConfigConstants';
+import { CHAIN_ID } from 'constants/chainConstants';
+
+// Config
+import { getEnv } from 'configs/envConfig';
+import { ARCHANOVA_RAMP_CURRENCY_TOKENS, ETHERSPOT_RAMP_CURRENCY_TOKENS } from 'configs/rampConfig';
+import { CONTAINER_ID, ORIGIN } from 'configs/wertConfig';
+
+// Services
+import etherspotService from 'services/etherspot';
+
+// Utils
+import { chainFromChainId } from 'utils/chains';
+
+const PILLAR = 'Pillar';
+
+export const buildUrlOptions = (options: { [key: string]: string }): string => {
+  let optionStr = '';
+  Object.keys(options).map((key: string) => {
+    let value = options[key];
+    optionStr += `${!optionStr ? '?' : '&'}${key}=${encodeURIComponent(value)}`;
+  });
+  return optionStr;
+};
+
+// RAMP
+export const rampWidgetUrl = (
+  address: string,
+  fiatCurrency: string = '',
+  fiatValue: string = '',
+  isEtherspotAccount: boolean = true,
+) => {
+  const params = {
+    hostAppName: PILLAR,
+    hostApiKey: getEnv().RAMPNETWORK_API_KEY,
+    userAddress: address,
+    fiatCurrency,
+    fiatValue,
+    swapAsset: isEtherspotAccount ? ETHERSPOT_RAMP_CURRENCY_TOKENS.join(',') : ARCHANOVA_RAMP_CURRENCY_TOKENS,
+  };
+
+  const url = `${getEnv().RAMPNETWORK_WIDGET_URL}/${buildUrlOptions(params)}`;
+  return url;
+};
+
+// WERT
+export const wertWidgetUrl = (address: string, fiatValue: string) => {
+  const wertWidget = new WertWidget({
+    partner_id: getEnv().WERT_ID,
+    container_id: CONTAINER_ID,
+    origin: ORIGIN,
+    commodities: firebaseRemoteConfig.getString(REMOTE_CONFIG.FEATURE_WERT_COMMODITIES),
+    currency: firebaseRemoteConfig.getString(REMOTE_CONFIG.FEATURE_WERT_CURRENCY),
+    currency_amount: parseFloat(fiatValue),
+    commodity: firebaseRemoteConfig.getString(REMOTE_CONFIG.FEATURE_WERT_COMMODITY),
+    address,
+  });
+
+  return wertWidget.getEmbedUrl();
+};
+
+// Pelerin
+export const buildMtPelerinOptions = (code: string, address: string) => {
+  let onRampOptions = {
+    lang: 'en',
+    tab: 'buy',
+    tabs: 'buy',
+    chain: 'xdai_mainnet',
+    net: 'matic_mainnet',
+    nets: 'arbitrum_mainnet,avalanche_mainnet,bsc_mainnet,fantom_mainnet,mainnet,optimism_mainnet,xdai_mainnet',
+    crys: 'AVAX,BNB,BTCB,BUSD,DAI,ETH,FRAX,LUSD,MAI,MATIC,RBTC,RDOC,RIF,USDC,USDT,WBTC,WETH,XCHF,XDAI,XTZ',
```

Comment:
that's a weird property name 😂

Reply:
lmao i've seen assets, tokens, even coins - but never crys, maybe the 2 saved chars make a difference